Test.Activities.Automation.ActivityLib/Models/Activity/Services/SyncActivityService.cs

```diff
@@ -0,0 +1,246 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.SharePoint;
+using Test.Activities.Automation.ActivityLib.Models.Activity.Classes;
+using Test.Activities.Automation.ActivityLib.Models.Helpers;
+using Test.Activities.Automation.ActivityLib.Utils.Constants;
+
+namespace Test.Activities.Automation.ActivityLib.Models.Activity.Services.SyncActivityService
+{
+    public class SyncActivityService
+    {
+        private ILogger _logger;
+
+        public SyncActivityService(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public void SyncActivities(IEnumerable<ActivityInfo> activities)
+        {
+            try
+            {
+                _logger?.LogInformation("Synchronizing activities");
+
+                if (activities == null)
+                {
+                    throw new Exception("Activities are null");
+                }
+
+                var now = DateTime.Now;
+                var minDate = new DateTime(now.Year, now.AddMonths(-1).Month, 1);
+                var maxDate = now.Date;
+
+                var ensureService = new EnsureService(_logger);
+
+                ensureService.CheckActivityDate(ref activities, minDate, maxDate);
```

Comment:
You implemented complex dictionary key in `EnsureService` which depends on month and year. But here you remove new activities which are for other monts. I'm not sure, but it seems like if you remove this line and `CheckActivityDate` method, code still works. Additionaly: `var minDate = activities.Min(x => x.Date);` and `var maxDate = activities.Max(x => x.Date);`